### PR TITLE
fix(wasm|droid): Textblock can be constrained by maxlines

### PR DESCRIPTION
**Pull request by alextrepa**
GitHub Issue (If applicable): Closing #3714

## PR Type

What kind of change does this PR introduce?
- Bugfix

## What is the current behavior?
Max lines was not working in wasm, except in a stack panel, and it was not working in android.

## What is the new behavior?
The max line works.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] ~Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)~
- [ ] ~[Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)~
- [ ] ~Validated PR `Screenshots Compare Test Run` results.~
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.wasm.cs [hunk 1]
```diff
@@ -45,6 +45,16 @@ public TextBlock() : base("p")
...
+			InitializeDefaultValues();
+
+		}
+
+		/// <summary>
+		/// Set default properties to align them with uwp.
+		/// </summary>
+		private void InitializeDefaultValues()
+		{
+			this.SetValue(VerticalAlignmentProperty, VerticalAlignment.Top, DependencyPropertyValuePrecedences.DefaultValue);
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.wasm.cs:57]: This is not correct, the default `VerticalAlignment` is `Stretch` for `TextBlock` as for other FrameworkElement types. Can you share more information on why you added this code?

File: src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.Android.cs [hunk 2]
```diff
@@ -156,7 +156,8 @@ private void Update()
 			UpdateTextTrimming();
 			UpdateTextFormatted();
 			UpdatePaint();
-		}
+            UpdateMaxLines();
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.Android.cs:159]: indentation

File: src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.Android.cs [hunk 3]
```diff
@@ -241,6 +242,22 @@ private void UpdateTextFormatted()
 			_textFormatted = GetTextFormatted();
 		}
 
+		/// <summary>
+		/// Sets ellipsize to truncateAt.END when a number of line is specified.
+		/// If ellipsize is another value, the text won't truncate.
+		/// </summary>
+		private void UpdateMaxLines()
+		{
+			if (_ellipsize != null)
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.Android.cs:251]: You should probably reset `_ellipsize` to `null` when `MaxLines` changes (similar to `TextTrimming`)

File: src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.cs [hunk 4]
```diff
@@ -635,6 +635,11 @@ private void OnTextDecorationsChanged()
 		/// </summary>
 		private bool UseInlinesFastPath => _inlines == null;
 
+		/// <summary>
+		/// Returns if the TextBlock is constrained by a maximum number of lines.
+		/// </summary>
+		private bool IsLayoutConstrained => MaxLines > 0;
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.cs:641]: I would call this `IsLayoutConstrainedByMaxLines`

**Review by davidjohnoliver — COMMENTED** [review]
This needs a UI test

File: src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.Android.cs (hunk 3 above)
Comment (alextrepa) [reply src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.Android.cs:251]: That is what I do, if it's not null, I return it, like in text trimming, if the layout is constrained, I set it to TruncateEnd, else I set it to null. What am I missing?

Comment (davidjohnoliver) [reply src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.Android.cs:251]: look at what happens when the `TextTrimming` property changes

File: src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.Android.cs (hunk 2 above)
Comment (alextrepa) [reply src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.Android.cs:159]: I know that it looks like the indentation is wrong here, but that is just the way github presents the information, if you go see the file, the indentation is fine.
https://github.com/unoplatform/uno/blob/85d547691ae52af9c50d98b00e9dd227389061f2/src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.Android.cs#L159

Comment (davidjohnoliver) [reply src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.Android.cs:159]: in fact it seems to be using spaces. Adjust it to use tabs.

**Review by carldebilly — APPROVED** [review]

File: src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/TextBlockControl/SimpleText_MaxLines_Mutiple_Containers.xaml.cs [hunk 5]
```diff
@@ -0,0 +1,16 @@
+﻿using Uno.UI.Samples.Controls;
+using Windows.UI.Xaml.Controls;
+
+
+namespace Uno.UI.Samples.Content.UITests.TextBlockControl
+{
+	[SampleControlInfo("TextBlockControl", "SimpleText_MaxLines_Mutiple_Containers", description: "This sample shows two different parents for a text block. The maximum number of lines they should have is defined by the slider.")]
```
Comment (carldebilly) [inline src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/TextBlockControl/SimpleText_MaxLines_Mutiple_Containers.xaml.cs:7]: New approach is to use simply `[Sample]` attribute. The name & category are infered from class name/namespace.

**Review by carldebilly — APPROVED** [review]

File: src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.Android.cs (hunk 3 above)
Comment (carldebilly) [reply src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.Android.cs]: I fixed this yesterday.

**Pull request merged**